### PR TITLE
Change AuthController to fix #53

### LINKED ISSUE #53: Security concerns over longevity of tokens
In today's meeting, the SoC systems engineer raised concerns over the security of the long-lived (52 weeks) refresh token. He recommends a validity period more similar to IVLE (8 hours).

### PR DESCRIPTION
**Pull request by indocomsoft**
Fixes #53

### CODE REVIEW

File: lib/cadet_web/controllers/auth_controller.ex [hunk 1]
```diff
@@ -56,24 +56,35 @@ defmodule CadetWeb.AuthController do
...
+  Receives a /refresh request with valid attribute.
+
+  Exchanges the refresh_token with a new access_token.
+  """
   def refresh(conn, %{"refresh_token" => refresh_token}) do
     case Guardian.exchange(refresh_token, "refresh", "access") do
-      {:ok, {refresh_token, _}, {access_token, _}} ->
-        render(conn, "token.json", access_token: access_token, refresh_token: refresh_token)
+      {:ok, {_refresh_token, _}, {access_token, _}} ->
+        render(conn, "access_token.json", access_token: access_token)
```
Comment (tuesmiddt) [inline lib/cadet_web/controllers/auth_controller.ex]: We're not giving them a new refresh token (sliding window behaviour)?

File: lib/mix/tasks/token.ex [hunk 2]
```diff
@@ -40,11 +40,16 @@ defmodule Mix.Tasks.Cadet.Token do
       user = test_user(role)
 
       {:ok, access_token, _} =
-        Guardian.encode_and_sign(user, %{}, token_type: "access", ttl: {4, :weeks})
+        Guardian.encode_and_sign(user, %{}, token_type: "access", ttl: {1, :day})
```
Comment (tuesmiddt) [inline lib/mix/tasks/token.ex]: Not sure what's the point of making this change tbh, outside of testing the auth controller but I guess it doesn't hurt either.

File: lib/cadet_web/controllers/auth_controller.ex (hunk 1 above)
Comment (indocomsoft) [reply lib/cadet_web/controllers/auth_controller.ex]: Oh, hmm okay I realise why we had this now. Will revert back to the old one then.

Also need to change from `exchange` to `encode_and_sign` then, as in totally issuing a new one. Will make the necessary changes.

**Comment by indocomsoft** [comment]
All fixed and rebased on `master` @tuesmiddt

File: lib/cadet_web/controllers/auth_controller.ex [hunk 3]
```diff
@@ -56,33 +52,60 @@ defmodule CadetWeb.AuthController do
...
+  Exchanges the refresh_token with a new access_token.
+  """
   def refresh(conn, %{"refresh_token" => refresh_token}) do
-    case Guardian.exchange(refresh_token, "refresh", "access") do
-      {:ok, {refresh_token, _}, {access_token, _}} ->
+    # TODO: Refactor to use refresh after guardian_db > v1.1.0 is released.
+    case Guardian.resource_from_token(refresh_token) do
+      {:ok, user, %{"typ" => "refresh"}} ->
+        %{access_token: access_token, refresh_token: refresh_token} = generate_tokens(user)
         render(conn, "token.json", access_token: access_token, refresh_token: refresh_token)
```
Comment (tuesmiddt) [inline lib/cadet_web/controllers/auth_controller.ex]: What do you think about just doing:

`render(conn, "token.json", generate_tokens(user))`
or
```
tokens = generate_tokens(user)
render(conn, "token.json", tokens)
```

Comment (indocomsoft) [reply lib/cadet_web/controllers/auth_controller.ex]: Oops. You're right. Will fix this

**Comment by indocomsoft** [comment]
Fixed! @tuesmiddt

**Pull request merged**